st2actions/st2actions/workflows/workflows.py

```diff
@@ -0,0 +1,85 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import kombu
+
+from orchestra import states
+
+from st2common import log as logging
+from st2common.models.db import workflow as wf_db_models
+from st2common.services import workflows as wf_svc
+from st2common.transport import consumers
+from st2common.transport import queues
+from st2common.transport import utils as txpt_utils
+
+
+LOG = logging.getLogger(__name__)
+
+
+WORKFLOW_EXECUTION_QUEUES = [
+    queues.WORKFLOW_EXECUTION_WORK_QUEUE,
+    queues.WORKFLOW_EXECUTION_RESUME_QUEUE
+]
+
+
+class WorkflowDispatcher(consumers.MessageHandler):
+    message_type = wf_db_models.WorkflowExecutionDB
+
+    def __init__(self, connection, queues):
+        super(WorkflowDispatcher, self).__init__(connection, queues)
+
+    def get_queue_consumer(self, connection, queues):
+        # We want to use a special ActionsQueueConsumer which uses 2 dispatcher pools
```

Comment:
Need updating now?

Reply:
That's a good question because the previous one was specifically for ActionRunners.  I caught this while troubleshooting the Xenial issue.  The use of the normal QueueConsumer seems to work as well.